pandas/io/parsers.py

```diff
@@ -384,6 +384,38 @@ def _validate_integer(name, val, min_val=0):
     return val
 
 
+def _check_dup_names(names):
+    """
+    Check if the `names` parameter contains duplicates.
+
+    Currently, this function issues a warning if that is the case. In the
+    future, we will raise an error.
+
+    Parameters
+    ----------
+    names : array-like or None
+        An array containing a list of the names used for the output DataFrame.
+    """
+
+    if names is None:
+        return
+
+    counts = {}
+    warn_dups = False
+
+    for name in names:
+        if name in counts:
+            warn_dups = True
+            break
+
+        counts[name] = True
+
+    if warn_dups:
+        msg = ("Duplicate names specified. This "
```

Comment:
so are we deprecating this? then this should be a FutureWarning.

Reply:
Fair enough.  Done.